src/utils/deprecated/deprecated.tsx

```diff
@@ -3,25 +3,65 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React, { useEffect } from 'react';
+import React, { useEffect, useRef } from 'react';
 
 export const getDeprecatedMessage = (message: string): string =>
   `[DEPRECATED] ${message}`;
 
-export const deprecated = (message: string) => {
+interface IDeprecatedComponentWarning {
+  newComponentName: string;
+  version?: string;
+  getMessage?: (deprecatedComponentName: string) => string;
+}
+
+export const deprecatedComponentWarning = ({
+  newComponentName,
+  version,
+  getMessage,
+}: IDeprecatedComponentWarning) => {
   return <T extends React.ComponentType<any>>(Component: T): T => {
+    const deprecatedComponentName = Component.displayName || Component.name;
+
     const DeprecatedWrapper = (props: React.ComponentProps<T>) => {
       useEffect(() => {
-        console.warn(getDeprecatedMessage(message));
+        const message =
+          getMessage?.(deprecatedComponentName) ||
+          `${deprecatedComponentName} is deprecated in favor of ${newComponentName} and will be removed in v${version}.`;
+        const deprecatedMessage = getDeprecatedMessage(message);
+
+        console.warn(deprecatedMessage);
       }, []);
 
       return <Component {...props} />;
     };
 
     Object.defineProperty(DeprecatedWrapper, 'name', {
-      value: Component.displayName || Component.name,
+      value: deprecatedComponentName,
     });
 
     return DeprecatedWrapper as T;
   };
 };
+
+interface IDeprecatedPropWarning {
+  props: Record<string, any>;
```

Comment:
```suggestion
  props: Record<string, unknown>;
```

Haven't tested if this suggestion actually works, but it would be ideal to avoid the usage of `any`

Reply:
It works fine, I've made changes